### ISSUE TITLE
If a font supports ligatures, 'smallcaps' doesn't seem to affect them

### ISSUE THREAD

**Issue by jsx97** [typst/typst#5313]
### Description

If a font supports ligatures, `smallcaps` doesn't seem to affect them.

```typ
#set text(font: "New Computer Modern")

#smallcaps[Before & After]
```

<img width="226" alt="image" src="https://github.com/user-attachments/assets/934e58f7-7417-4010-9aaf-1cf10f520b74">

If I understand correctly, if a font doesn't support ligatures, this issue doesn't occur:

```typ
// #set text(font: "New Computer Modern")  // We use the default Typst font instead

#smallcaps[Before & After]
```

<img width="215" alt="image" src="https://github.com/user-attachments/assets/1d5543a7-8195-4938-a035-578e8c0745c6">

### Operating system

macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by EpicEricEE** [comment]
This is a font issue, as the font should order the lookup tables of the different features correctly. I verified this by moving the lookup tables in the font's source around, which fixed this issue.

Currently, the ligature table comes first, so the "f​‌ t" is replaced by the ligature "ft". When it then gets to the small-caps table, it can't find a replacement for that ligature character, so it stays as is. If the font gave higher priority to the small-caps table, they will be applied first, so that only ligatures meant for small-caps characters are applied later.

**Comment by laurmaedje** [comment]
@EpicEricEE IIRC correctly you've been in contact with the maintainer of New CM before, right? Would you mind reporting this upstream?

**Comment by EpicEricEE** [comment]
I have reported it (and #5126), the next font update should fix them both.

**Comment by laurmaedje** [comment]
Great, thanks!

**Comment by EpicEricEE** [comment]
He released version 7.0.0, which should be available on CTAN soon (including a new sans math font).
